ui/src/segments/components/form/validation/schema.js

```diff
@@ -2,6 +2,62 @@
 import * as yup from "yup";
 
 const segmentNameRegex = /^[A-Za-z\d][\w\d \-()#$%&:.]*[\w\d\-()#$%&:.]$/;
+export const segmentConfigSchema = yup.lazy((obj) =>
+  yup.object().shape(
+    Object.keys(obj).reduce((acc, key) => {
+      return {
+        ...acc,
+        [key]: yup.mixed()
+          .when("$segmenterTypes", (segmenterTypes) => {
+            switch ((segmenterTypes[key] || "").toUpperCase()) {
+              case "BOOL":
+                return yup.array(
+                  yup
+                    .bool()
+                    .typeError("Array elements must all be of type: BOOL")
+                );
+              case "INTEGER":
+                return yup.array(
+                  yup
+                    .number()
+                    .integer()
+                    .typeError(
+                      "Array elements must all be of type: INTEGER"
+                    )
+                );
+              case "REAL":
+                return yup.array(
+                  yup
+                    .number()
+                    .typeError("Array elements must all be of type: REAL")
+                );
+              case "STRING":
+                return yup.array(
+                  yup
+                    .string()
+                    .typeError("Array elements must all be of type: STRING")
+                );
+              default:
+                return yup.array(); // Type is unknown for deactivated segmenters
+            }
+          })
+          .when(
```

Comment:
Previously, the `.when("$segmenterTypes", (segmenterTypes)` check came after this and if so, the chaining of checks will only work if we do `schema.array()` as opposed to a new `yup.array()`. Anyway, I've now moved the type check above instead.